### PR TITLE
Improve diagnostics of Controllers, Hardware Components and Controller Manager

### PR DESCRIPTION
**Pull request by saikishor**
This PR in general aims to improve the diagnostics of the ros2_control

### CODE REVIEW

**Review by christophfroehlich — DISMISSED** [review]
LGTM

File: controller_manager/src/controller_manager.cpp [hunk 1]
```diff
@@ -2748,13 +2764,95 @@ void ControllerManager::controller_activity_diagnostic_callback(
...
+  {
+    stat.summary(
+      diagnostic_msgs::msg::DiagnosticStatus::ERROR, "No hardware components are loaded!");
+  }
+  else
+  {
+    if (!atleast_one_hw_active)
+    {
+      stat.summary(
+        diagnostic_msgs::msg::DiagnosticStatus::ERROR,
```
Comment (destogl) [inline controller_manager/src/controller_manager.cpp:2822]: I would go here with warn acutally.

**Review by destogl — APPROVED** [review]

**Review by destogl — DISMISSED** [review]
Small change,  otherwise good.

**Review by bmagyar — APPROVED** [review]
Thank you!

**Pull request merged**